File: app/src/main/java/com/example/braincard/data/model/Deck.kt
package com.example.braincard.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "Deck", foreignKeys =
    [ForeignKey(entity = Gruppo::class, parentColumns = ["id"], childColumns = ["idGruppo"], onDelete = ForeignKey.CASCADE)]
)
data class Deck (
    @PrimaryKey val id : String = "",
    var nome:String="",
    var percentualeCompletamento : Int = 0,
    var idGruppo:String=""


)

File: FLUTTER/braincard_flutter/lib/model/deck.dart
class Deck {
  final String id;
  final String nome;
  int percentualeCompletamento;
  final String idGruppo;

  Deck({
    required this.id,
    required this.nome,
    required this.percentualeCompletamento,
    required this.idGruppo,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'nome': nome,
      'percentualeCompletamento': percentualeCompletamento,
      'idGruppo': idGruppo,
    };
  }

  factory Deck.fromMap(Map<String, dynamic> map) {
    return Deck(
      id: map['id'],
      nome: map['nome'],
      percentualeCompletamento: map['percentualeCompletamento'],
      idGruppo: map['idGruppo'],
    );
  }
}
